lib/src/values/logic_value.dart

```diff
@@ -565,6 +565,58 @@ abstract class LogicValue {
     }
   }
 
+  /// Compares this to `other`.
+  ///
+  /// Returns a negative number if `this` is less than `other`, zero if they are
+  /// equal, and a positive number if `this` is greater than `other`.
+  ///
+  /// It will throw an exception if `this` or [other] is not [isValid] or for
+  /// non-equal `LogicValue` [width].
+  @override
+  int compareTo(Object other) {
+    if (other is! LogicValue) {
+      throw TypeError();
```

Comment:
Perhaps we should extend this or something, or go back to an exception, because this has no helpful message

Reply:
Adding `ValueTypeError` to the exceptions folder by extending `TypeError` looks like a good option but `ValueType`Error may confuse the developers. So going back to Exception() seems a good option, but let me know if `ValueTypeError() ` with message serves better in this case.